flink-python/setup.py

```diff
@@ -321,12 +321,11 @@ def find_file_path(pattern):
         license='https://www.apache.org/licenses/LICENSE-2.0',
         author='Apache Software Foundation',
         author_email='dev@flink.apache.org',
-        python_requires='>=3.5',
-        install_requires=['py4j==0.10.8.1', 'python-dateutil==2.8.0', 'apache-beam==2.23.0',
+        python_requires='>=3.6',
+        install_requires=['py4j==0.10.8.1', 'python-dateutil==2.8.0', 'apache-beam==2.27.0',
                           'cloudpickle==1.2.2', 'avro-python3>=1.8.1,<=1.9.1', 'jsonpickle==1.2',
-                          'pandas>=0.24.2,<1; python_full_version < "3.5.3"',
-                          'pandas>=0.25.2,<1; python_full_version >= "3.5.3"',
-                          'pyarrow>=0.15.1,<0.18.0', 'pytz>=2018.3', 'numpy>=1.14.3,<1.20'],
+                          'pandas>=0.25.2,<1', 'pyarrow>=0.15.1,<0.18.0', 'pytz>=2018.3',
+                          'numpy>=1.14.3,<1.20', 'fastavro>=0.21.4,<0.24'],
```

Comment:
```suggestion
                          'numpy>=1.14.3,<1.21', 'fastavro>=0.21.4,<0.24'],
```

Reply:
`fastavro` hasn't no longer provided manylinux2010.whl since `0.24.2`, which makes some old version machines must have gcc if using newer version of `fastavro`. In fact, the `fastavro` dependency comes from `apache-beam`, and it is used in connectors. We can limit it to a relatively old version to solve the problem of `gcc` absence, and it will not have any impact on Pyflink. What do you think?

The 2.0.0 version of `pyarrow` is incompatible with the 1.20.x version of `numpy`. This problem was resolved in the unreleased 2.0.1 version of `pyarrow`. So here we limit the version of `numpy` to less than 1.20.
https://issues.apache.org/jira/browse/ARROW-10855?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel